include/tvm/tir/stmt_functor.h

```diff
@@ -413,6 +413,16 @@ inline T Substitute(T input, const std::unordered_map<const VarNode*, PrimExpr>&
  */
 TVM_DLL void PreOrderVisit(const ObjectRef& stmt_or_expr,
                            const std::function<bool(const ObjectRef&)>& fvisit);
+
+class PrimFunc;
+/*!
+ * \brief Re-generate the definition nodes for a TIR, including VarDef, BufferDef.
+ *        This pass works as a simple DeepCopy to duplicate a function with different Vars and
+ *        Buffers but the same behavior
+ * \param func The input PrimFunc.
+ * \return The new generated func.
+ */
+TVM_DLL PrimFunc ReGenerateDef(const PrimFunc& func);
```

Comment:
I found the name `ReGenerateDef` very confusion for two reasons:
- ReGenerate usually means adding new things, while we are replacing variables;
- Def doesn't refer to specific definition...Without reading the comments, it's non-trivial to figure out what "def" are we regenerating...

I can quickly come up with a few alternatives, but have to admit i'm not good at naming...
* RenewVarBufferDef
* UpdateVarBufferDef
* SubstituteVarBufferDef

Reply:
How about Clone?

Reply:
Renew seems to be right. However the purpose here is to regenerate all defs, so RenewDefs and add comment that we are going to renew all defs including buffer/var(or future var like defs) sounds about right